neurostuff/NiMARE
# [ENH] add images_to_coordinates

Closes #423 .


Changes proposed in this pull request:

- use transformer class to make ImageToCoordinates
- test ImageToCoordinates
- add notebook demonstrating use of ImageToCoordinates
- add `z_to_p` function for cases when z-maps are not available, but you still specify `z_threshold` as a z-value
- test `z_to_p`
- fix bug in ` dict_to_coordinates`


- [x] How should I handle circular dependencies for `get_masker`?
  - have a round about somewhat hacky solution
- [x] how to handle missing required stat maps?
  - I decided to add a `merge_strategy` parameter with three different options for users to decide whether to overwrite existing coordinates, and to skip generating coordinates for study contrasts without the required z or p maps.
- [x] somewhat related to the above point, if I include peak statistics for the coordinates that are generated, how should the coordinates that already exist (and do not have peak statistics reported) be represented? As `NaN`s?
  - Representing missing peak coordinates as `NaN`s is my current solution.
- [x] how should negative and positive statistics be handled?
  - there is `two_sided` parameter now available thanks to @tsalo 
- [x] how should I represent metadata
  - have a `coordinate_source` column with either `original` or `nimare` (or None)
